addons/mrp/models/mrp_production.py

```diff
@@ -1134,7 +1134,8 @@ def action_generate_serial(self):
         self.ensure_one()
         self.lot_producing_id = self.env['stock.production.lot'].create({
             'product_id': self.product_id.id,
-            'company_id': self.company_id.id
+            'company_id': self.company_id.id,
+            'name': self.env['stock.production.lot'].get_next_serial(self.company_id, self.product_id) or "0000001",
```

Comment:
I would instead correct get_next_serial:
1. Make it private if not call by public method.
2. Add a domain in the search and a args for the picking_type. Since you can define a sequence by picking.type, it should not be override by a magic search (to check with mgm). I would instead look for the last serial with that picking type.
3. If we couldn't find a serial, call the picking type sequence to get the next serial number to generate. More robust than 000001